internal/controller/ols_app_postgres_reconciliator.go

```diff
@@ -81,146 +82,155 @@ func (r *OLSConfigReconciler) reconcilePostgresDeployment(ctx context.Context, c
 		r.logger.Info("creating a new OLS postgres deployment", "deployment", desiredDeployment.Name)
 		err = r.Create(ctx, desiredDeployment)
 		if err != nil {
-			return fmt.Errorf("%s: %w", ErrCreatePostgresDeployment, err)
+			return "Failed", fmt.Errorf("%s: %w", ErrCreatePostgresDeployment, err)
 		}
-		return nil
+
+		// Deployment was just created – it cannot be ready yet.
+		return "In Progress", fmt.Errorf("deployment is not ready, 0 replicas available")
```

Comment:
I doubt a deployment progressing to running phase is considered an error. It is expected behavior. we can check the returned message and queue a follow-up reconciliation several seconds later in the main reconciliation loop.

Moreover, a deployment just get updated or created has no chance to achieve ready state during the operator function call that only lasts several miliseconds. 

I suggest we check all the deployments' statuses at the end of main reconciliation loop and set the deployment status accordingly.

Reply:
We do not report this as an error. Its in Progress

Reply:
it is returning an error, caught by the outer loop, it will interrupt the reconciliation.